test/column/renderEditCell.test.tsx

```diff
@@ -71,17 +71,20 @@ describe('Editor', () => {
   it('should commit quickly enough on outside clicks so click event handlers access the latest rows state', async () => {
     const onSave = vi.fn();
     render(<EditorTest onSave={onSave} />);
-    await userEvent.dblClick(getCellsAtRowIndex(0)[0]);
-    await userEvent.keyboard('234');
+    const user = userEvent.setup();
+    await user.dblClick(getCellsAtRowIndex(0)[0]);
+    await user.keyboard('234');
     expect(onSave).not.toHaveBeenCalled();
     const saveButton = screen.getByRole('button', { name: 'save' });
-    fireEvent.mouseDown(saveButton);
+
     // await userEvent.click() triggers both mousedown and click, but without delay,
     // which isn't realistic, and isn't enough to trigger outside click detection
+    await user.pointer([{ keys: '[MouseLeft>]', target: saveButton }]);
     await act(async () => {
       await new Promise(requestAnimationFrame);
     });
```

Comment:
questions
1. is this `act` for saving animation?
2. how did we ensure that the commit is fast enough?

Reply:
1. The `act` is to wrap the other [`requestAnimationFrame`](https://github.com/adazzle/react-data-grid/blob/d5120d82e54548f80280d907166e97b76839e312/src/EditCell.tsx#L73) call from the grid
2. we test that `onSave` was called with updated data with this assertion:
   ```ts
     expect(onSave).toHaveBeenCalledWith([
       { col1: 2341, col2: 'a1' },
   ```

Reply:
thanks for the answer. I think I was curious on how RDG implemented for 2. but the answer of 1. helped me understand